### PR TITLE
Take maintenance mode into account while choosing coordinators and read maintenance mode information from special key space

### PR DESCRIPTION
**Pull request by sbodagala**
# Description

Extend the logic that selects/verifies the validity of current coordinators to take maintenance mode information into account.

Read maintenance mode information from special keyspace.

## Type of change

- Other (extends the existing functionality)

## Discussion

*Are there any design details that you would like to discuss further?*

No.

## Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

Ran the test, locally, that has been added in this PR.

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

Yes (nightly regression tests will need to be run).

## Documentation

*Did you update relevant documentation within this repository?*

No.

*If this change is adding new functionality, do we need to describe it in our user manual?*

Maybe (at a later point).

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

No (maybe at a later point).

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

N/A (I think).

## Follow-up

*Are there any follow-up issues that we should pursue in the future?*

No.

*Does this introduce new defaults that we should re-evaluate in the future?*

No.

### CODE REVIEW

File: api/v1beta2/foundationdb_status.go [hunk 1]
```diff
@@ -134,6 +134,8 @@ type FoundationDBStatusProcessInfo struct {
 	// Excluded indicates whether the process has been excluded.
 	Excluded bool `json:"excluded,omitempty"`
 
+	UnderMaintenance bool `json:"under_maintenance,omitempty"`
```
Comment (johscheuer) [inline api/v1beta2/foundationdb_status.go:137]: Can we add a comment above the new variable describing what it reflects?

**Review by johscheuer — COMMENTED** [review]
Changes LGTM 👍 Can you please add a comment on the new variable. Do you think it makes sense to add an e2e test case for this into the operator test suite to make sure the system behaves correct?

**Comment by sbodagala** [comment]
>  Do you think it makes sense to add an e2e test case for this into the operator test suite to make sure the system behaves correct?

Yes. How about we do that in a follow up radar/PR (that allows this PR to be merged and also avoids the need to make an image with these changes?

**Review by johscheuer — APPROVED** [review]
>> Do you think it makes sense to add an e2e test case for this into the operator test suite to make sure the system behaves correct?
> Yes. How about we do that in a follow up radar/PR (that allows this PR to be merged and also avoids the need to make an image with these changes)?

That's fine for me but you don't have to build any images, the e2e pipeline will build the operator image based on the changes in the current branch e.g. would include your changes.

**Comment by sbodagala** [comment]
> > > Do you think it makes sense to add an e2e test case for this into the operator test suite to make sure the system behaves correct?
> > > Yes. How about we do that in a follow up radar/PR (that allows this PR to be merged and also avoids the need to make an image with these changes)?
> 
> That's fine for me but you don't have to build any images, the e2e pipeline will build the operator image based on the changes in the current branch e.g. would include your changes.

No, I meant if we like to run a real-cluster test locally (over a PR that includes changes to the operator code) we will need to build an image?

**Comment by sbodagala** [comment]
Ok, let's merge this PR and write the test in a follow up PR. Thanks!

**Comment by johscheuer** [comment]
> > > > Do you think it makes sense to add an e2e test case for this into the operator test suite to make sure the system behaves correct?
> > > > Yes. How about we do that in a follow up radar/PR (that allows this PR to be merged and also avoids the need to make an image with these changes)?
> > 
> > 
> > That's fine for me but you don't have to build any images, the e2e pipeline will build the operator image based on the changes in the current branch e.g. would include your changes.
> 
> No, I meant if we like to run a real-cluster test locally (over a PR that includes changes to the operator code) we will need to build an image?

The e2e test are using real clusters, so you're talking about doing manual testing?

**Comment by sbodagala** [comment]
Integration checks failed above, not sure what the problem is though. For example, https://github.com/FoundationDB/fdb-kubernetes-operator/actions/runs/5147753153/jobs/9268556581?pr=1652 says "Process exited with error code", not sure what that really means.

**Comment by sbodagala** [comment]
> ### Result of fdb-kubernetes-operator-pr on Linux CentOS 7
> * Commit ID: [c319531](https://github.com/FoundationDB/fdb-kubernetes-operator/commit/c319531591dfddce6ed0a478ca1c76c94945620a)
> * Duration 3:00:13
> * Result: ❌ **FAILED**
> * Error: `Error while executing command: if $fail_test; then exit 1; fi. Reason: exit status 1`
> * [Build Log](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator/fdb-kubernetes-operator-pr/0b0c6f2b-02c5-4552-a4e8-43096acbfc4a.gz) terminal output (available for 30 days)
> * [Build Workspace](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator-pr/0b0c6f2b-02c5-4552-a4e8-43096acbfc4a/workspace.zip) zip file of the working directory (available for 30 days)

This is because of this failure:

> TESTS FAILED SEE THESE LOGS:
> 
> /codebuild/output/src147849312/src/github.com/FoundationDB/fdb-kubernetes-operator/logs/test_operator.log
> 

In test ""maintenance mode is on" the cluster status is returning an empty maintenance mode string, causing the test to fail.

**Comment by sbodagala** [comment]
Disabled the maintenance related test, so this PR can be merged. I will investigate the test failure and re-enable it in a follow up PR.

**Comment by johscheuer** [comment]
> Integration checks failed below, not sure what the problem is though. For example, https://github.com/FoundationDB/fdb-kubernetes-operator/actions/runs/5147753153/jobs/9268556581?pr=1652 says "Process exited with error code", not sure what that really means.

Could

**Pull request merged**